src/server/Server.java

```diff
@@ -9,92 +15,72 @@
-import java.util.Map.Entry;
 import java.util.concurrent.ArrayBlockingQueue;
 import java.util.concurrent.RejectedExecutionException;
 import java.util.concurrent.ThreadPoolExecutor;
 import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
 
-public abstract class Server {
-    private final SparkDB MIME = new SparkDB();
-    private final HashMap<String, String> CustomHeaders = new HashMap<>();
-    private int Port = 80;
-    private int MaxConcurrentRequests = 1000;
-    private boolean GZip = true;
-    private int MaxRequestSizeKB = 100_000;
-    private int backlog = MaxConcurrentRequests * 5;
-    private String MIMEFile = "etc/MIME.db";
-    private String WWWDir = "www";
-    private int SOTimeout = 60_000;
-    private int BufferSize = 524288;
+public class Server {
+    public static final String HTTP_PROTO_VERSION = "HTTP/1.1";
 
-    private static String getStackTrace(final Throwable throwable) {
-        final StringWriter sw = new StringWriter();
-        final PrintWriter pw = new PrintWriter(sw, true);
-        throwable.printStackTrace(pw);
-        return sw.getBuffer().toString();
-    }
-
-    public void setSOTimeout(int in) {
-        SOTimeout = in;
-    }
-
-    public void setBufferSize(int in) {
-        BufferSize = in;
-    }
-
-    public void setMaximumConcurrentRequests(int in) {
-        MaxConcurrentRequests = in;
-        backlog = MaxConcurrentRequests * 5;
-    }
+    public final int port;
+    public final int maxRequestSizeBytes;
+    public final int maxConcurrentRequests;
+    public final int backlog;
+    public final boolean useGzip;
+    public final String rootDir;
+    public final String mimeFile;
 
-    public void setGZip(boolean in) {
-        GZip = in;
-    }
+    private final SparkDB MIME = new SparkDB();
+    private final Function<HttpRequest, HttpResponse> handler;
 
-    public void setMaximumRequestSizeInKB(int in) {
-        MaxRequestSizeKB = in;
-    }
 
-    public void setBacklog(int in) {
-        backlog = in;
+    public Server(Function<HttpRequest, HttpResponse> handler) {
+        this(8080, false, handler);
     }
 
-    public void setMIMEFile(String in) throws Exception {
-        MIMEFile = in;
-        loadMIME();
+    public Server(int port, boolean useGzip, Function<HttpRequest, HttpResponse> handler) {
+        this(port, 100000, 4096, 4096, useGzip, handler);
     }
 
-    public void setWWWDirectory(String in) {
-        WWWDir = in;
+    public Server(int port, int maxConcurrentRequests, int maxRequestSizeBytes, int maxResponseSizeBytes, boolean useGzip, Function<HttpRequest, HttpResponse> handler) {
+        this(port, maxConcurrentRequests, maxRequestSizeBytes, useGzip, "www", "etc/MIME.db", handler);
     }
 
-    public void setPort(int in) {
-        Port = in;
+    public Server(int port, int maxConcurrentRequests, int maxRequestSizeBytes, boolean useGzip, String rootDir, String mimeFile, Function<HttpRequest, HttpResponse> handler) {
```

Comment:
Setters are better than constructors

Reply:
No, No one will need to change the port of the server after starting the it, and most of the configurations need to be configured once at the begining.